src/drivers/display/ft81x/lv_ft81x.c

```diff
@@ -0,0 +1,435 @@
+    lv_ft81x_write_32(disp, EVE_RAM_DL, DL_CLEAR_RGB);
+    lv_ft81x_write_32(disp, EVE_RAM_DL + 4, (DL_CLEAR | CLR_COL | CLR_STN | CLR_TAG));
+    lv_ft81x_write_32(disp, EVE_RAM_DL + 8, DL_DISPLAY);    /* end of display list */
+    lv_ft81x_write_32(disp, REG_DLSWAP, EVE_DLSWAP_FRAME);
+
+    /* nothing is being displayed yet... the pixel clock is still 0x00 */
+    lv_ft81x_write_8(disp, REG_GPIO,
+                     0x80); /* enable the DISP signal to the LCD panel, it is set to output in REG_GPIO_DIR by default */
+    lv_ft81x_write_8(disp, REG_PCLK, params->pclk); /* now start clocking data to the LCD panel */
+
+    LV_ASSERT(lv_ft81x_read_16(disp, REG_CMD_READ) != 0xfff);
+
+    LV_ASSERT(0 == lv_ft81x_read_16(disp, REG_CMD_WRITE));
+
+
+    lv_ft81x_cmd_list_t cmd_list;
+    lv_ft81x_cmd_list_init(disp, &cmd_list);
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, CMD_MEMSET);
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, 0); /* address */
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, 0x00); /* val */
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, 2 * params->hor_res * params->ver_res); /* count */
+    lv_ft81x_cmd_list_send(disp, &cmd_list);
+
+    lv_ft81x_cmd_list_init(disp, &cmd_list);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, CMD_DLSTART);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, DL_CLEAR_RGB | 0); /* clear to black */
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, DL_CLEAR | CLR_COL | CLR_STN | CLR_TAG);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, TAG(0));
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, TAG(20));
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, CMD_SETBITMAP);
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, 0); /* address */
+    lv_ft81x_cmd_list_add_16(disp, &cmd_list, EVE_RGB565);
+    lv_ft81x_cmd_list_add_16(disp, &cmd_list, params->hor_res);
+    lv_ft81x_cmd_list_add_16(disp, &cmd_list, params->ver_res);
+    lv_ft81x_cmd_list_add_16(disp, &cmd_list, 0);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, DL_BEGIN | EVE_BITMAPS);
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, VERTEX2F(0, 0));
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, DL_END);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, TAG(0));
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, DL_DISPLAY);
+
+    lv_ft81x_cmd_list_add_32(disp, &cmd_list, CMD_SWAP);
+
+    lv_ft81x_cmd_list_send(disp, &cmd_list);
+
+
+    return LV_RESULT_OK;
+}
+
+static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map)
+{
+    lv_ft81x_driver_data_t * drv = lv_display_get_driver_data(disp);
+
+    int32_t hor_res = lv_display_get_horizontal_resolution(disp);
+    uint32_t disp_row_bytes = hor_res * 2;
+
+    uint32_t address = disp_row_bytes * area->y1 + area->x1 * 2;
+    uint8_t encoded_address[3];
+
+    if(lv_area_get_width(area) == hor_res) {
+        lv_ft81x_encode_write_address(encoded_address, address);
+        drv->spi_cb(disp, LV_FT81X_SPI_OPERATION_CS_ASSERT, NULL, 0);
```

Comment:
I know that the flush callback should be performant, but I think we should  validate if a callback is set before executing it and put a LOG_ERR to make easier to debug silent  corrupted drivers.